core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -675,7 +675,7 @@ class DAGScheduler(
     // Cancel all jobs belonging to this job group.
     // First finds all active jobs with this group id, and then kill stages for them.
     val activeInGroup = activeJobs.filter(activeJob =>
-      groupId == activeJob.properties.get(SparkContext.SPARK_JOB_GROUP_ID))
+      Option(activeJob.properties).exists(_.get(SparkContext.SPARK_JOB_GROUP_ID) == groupId))
```

Comment:
I think you are right, for the current codes, activeJob.properties will always be non-null.  But the codes may change in the future, and the above line make sure there won't be NPE even if activeJob.properties is null.